callbackify/index.ts

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as is from 'is';
```

Comment:
A few general thoughts.  

Do we need our own implementation of callbackify?  Starting with Node.js 8 and up, there is `util.callbackify`.  If the answer is to wait for the node.js 6 deprecation so we don't need to shim the world, that could be fine.  I'd just prefer to not introduce this if we don't have to :)

Second thing - funny as it sounds, the answer may be to put this in the `@google-cloud/promisify` npm module instead of adding it directly to the client lib.  We should have this in a generalized place.

Reply:
I only added `callbackify` module locally here to show/test how refactored async-first function would look like with `callbackifyAll` application. Once it is approved here, I will submit PR to the appropriate repo. 

We still need `callbackifyAll` and once node.js 6 is deprecated, swap the method to `util.callbackify`.

The use `callbackifyAll` 
* eliminates the need for "hybrid" params like `optionsOrCallback` and accompanying decryption logic.  
* cleans up a great deal of the `return` logic of the function.
from 
   ```javascript
    // tslint:disable-next-line no-any
    let resp: LogSink|any = null;
    let error: Error|null = null;
    let sink: Sink|null = null;
    try {
      [resp] = await this.configService.createSink(reqOpts, config.gaxOptions);
      sink = self.sink(resp.name);
      sink.metadata = resp;
    } catch (err) {
      error = err;
    }

    if (callback) {
      return callback(error, sink, resp);
    }
    if (error) {
      throw error;
    }
    return [sink as Sink, resp as LogSink];
  }
   ```
   to just
   ```javascript
    const [resp] =
        await this.configService.createSink(reqOpts, config.gaxOptions);
    const sink = self.sink(resp.name);
    sink.metadata = resp;
    return [sink as Sink, resp];
   ```

Reply:
@JustinBeckwith out of curiosity what is your stance on TypeScript decorators? Seems like we might be able to forgo a `callbackifyAll` method if we just decorated each method that needed it.